src/pages/content-edit/schedule-box.js

```diff
@@ -0,0 +1,143 @@
+import m from "mithril";
+import formatDate from "date-fns/format";
+
+import css from "./head.css";
+
+var DEFAULT_START_TIME = "00:00",
+    DEFAULT_END_TIME   = "23:59",
+    DATE_FORMAT = "YYYY-MM-DD",
+    TIME_FORMAT = "HH:mm",
+    TIMESTAMP_FORMAT = "x";
+
+function scheduleStr(side, date, time) {
+    if(!date) {
+        return null;
+    }
+
+    if(!time) {
+        time = side === "start" ? DEFAULT_START_TIME : DEFAULT_END_TIME;
+    }
+
+    return date + " " + time;
+}
+
+function timestampFromStr(str) {
+    return parseInt(formatDate(str, TIMESTAMP_FORMAT), 10);
+}
+
+function timestamp(side, date, time) {
+    var str = scheduleStr(side, date, time);
+
+    return str ? timestampFromStr(str) : null;
+}
+
+
+// All data in and out of the `scheduler` is in the form of timestamps.
+// All transformations to and from the date+time input fields will be
+// handled by this controller.
+
+export function controller(options) {
+    var ctrl = this,
+        content = options.content;
+
+    ctrl.content = content;
+
+    ctrl.inputs = null;
+    ctrl.ts = null;
+
+    ctrl.init = function() {
```

Comment:
It was a pattern I found myself using repeatedly so I could create floating "private" functions wherever they made sense -- e.g. alongside the functions that will use it -- without worrying about the order in which I was calling member variables as it related to eventually-hoisted functions. Calling .init at the end of a controller means you don't need to worry about the order in which things are declared.

Feel free to disagree on as many points as you'd like, of course. In this particular case, the pattern was not required by the final shape of the controller -- though an earlier iteration likely benefited from the init-last pattern.